crates/error/src/error.rs

```diff
@@ -0,0 +1,1751 @@
+///
+/// ```
+/// # use wasmtime_internal_error as wasmtime;
+/// use wasmtime::Error;
+///
+/// let messages = ["yikes", "uh oh", "ouch"];
+/// let errors = messages
+///     .into_iter()
+///     .map(Error::msg)
+///     .collect::<Vec<_>>();
+/// ```
+///
+/// # Printing an `Error`
+///
+/// Different format strings will print an `Error` differently:
+///
+/// * `{}`: Prints the `Display` of just the first error, without any of the
+///   other errors in the chain or the root cause.
+///
+/// * `{:#}`: Prints the `Display` of the first error, then (if there are more
+///   errors in the chain) a colon, then the display of the second error in the
+///   chain, etc...
+///
+/// * `{:?}`: Prints the `Display` of the first error, then (if there are more
+///   errors in the chain) a newline-separated list of the rest of the errors in
+///   the chain, and finally (if the `"backtrace"` cargo feature is enabled, the
+///   `RUST_BACKTRACE` environment variable is set and non-zero, and the
+///   platform is supported by Rust's standard library's `Backtrace` type) the
+///   captured backtrace is printed.
+///
+///   This is the default formatting used when `fn main() ->
+///   wasmtime::Result<()>` returns an error.
+///
+/// * `{:#?}`: Prints an internal, debugging representation of the `Error`. We
+///   make no guarantees about its stability.
+///
+/// Here is an example showing the different formats for the same error:
+///
+/// ```
+/// # fn _foo() {
+/// #![cfg(all(feature = "backtrace", not(miri)))]
+/// # let _ = unsafe { std::env::set_var("RUST_BACKTRACE", "1") };
+/// # use wasmtime_internal_error as wasmtime;
+/// use wasmtime::{bail, Context as _, Result};
+///
+/// fn uno() -> Result<()> {
+///     bail!("ouch")
+/// }
+///
+/// fn dos() -> Result<()> {
+///     uno().context("whoops")
+/// }
+///
+/// fn tres() -> Result<()> {
+///     dos().context("uh oh")
+/// }
+///
+/// let error = tres().unwrap_err();
+///
+/// println!("{error}");
+/// // Prints:
+/// //
+/// //     uh oh
+///
+/// println!("{error:#}");
+/// // Prints:
+/// //
+/// //     uh oh: whoops: ouch
+///
+/// println!("{error:?}");
+/// // Prints
+/// //
+/// //     uh oh
+/// //
+/// //     Caused by:
+/// //         0: whoops
+/// //         1: ouch
+/// //
+/// //     Stack backtrace:
+/// //       <...>
+/// //        7: example::uno
+/// //        8: example::dos
+/// //        9: example::tres
+/// //       10: example::main
+/// //       <...>
+///
+/// println!("{error:#?}");
+/// // Prints
+/// //
+/// //     Error {
+/// //         <...>
+/// //     }
+/// # }
+/// ```
+///
+/// # Converting a `wasmtime::Error` into an `anyhow::Error`
+///
+/// When the `"anyhow"` feature is enabled, there is a `From<wasmtime::Error>
+/// for anyhow::Error` implementation. You can always call that implementation
+/// explicitly if needed, but `?`-propagation allows the conversion to happen
+/// seemlessly from functions that return a `Result<T, wasmtime::Error>` to
```

Comment:
s/seemlessly/seamlessly/

Reply:
Well the conversion does happen without anyone seeming to have done anything :-p